### ISSUE TITLE
all.sh: psa-crypto: Prepare and clean out-of-tree build directory

### ISSUE THREAD

**Issue by ronald-cron-arm** [Mbed-TLS/mbedtls#8299]
### Suggested enhancement
In psa-crypto, we plan to build out-of-tree for most if not all all.sh components. To avoid having to set-up and clean-up the build directory in each component, do it in psa-crypto case systematically before and after running a component. 

Something along the lines of:
```
    PSA_CRYPTO_ROOT_DIR="$PWD"                                                  
    mkdir "$OUT_OF_SOURCE_DIR"                                                  
    cd "$OUT_OF_SOURCE_DIR" 
```
in run_component()?

Something along the lines of:
```
    cd "$PSA_CRYPTO_ROOT_DIR"                                                   
    rm -rf "$OUT_OF_SOURCE_DIR"
```
in cleanup()?

### Justification
 To avoid having to set-up and clean-up the build directory in each component.

**Comment by gilles-peskine-arm** [comment]
Makes sense. Each component should have its own build directory name so that you can run them in parallel. And the components that do in-tree builds should be indicated specially.

>  we plan to build out-of-tree for most if not all all.sh components

It can't be all because we need at least one test of in-tree builds for each build system.

Do you have support for out-of-tree builds with make (yay!) or are you planning to use cmake almost everywhere in all.sh (boo!)?

**Comment by ronald-cron-arm** [comment]
> Do you have support for out-of-tree builds with make (yay!) or are you planning to use cmake almost everywhere in all.sh (boo!)?

This is in the first place the TF-PSA-Crypto repo and it is CMake only.

**Comment by ronald-cron-arm** [comment]
We have not done exactly that but that part of the work to adapt all.sh to TF-PSA-Crypto is completed now.
